TunnelManager/Core.cs

```diff
@@ -57,7 +58,29 @@ private void Initialize()
         {
             foreach (string sessionName in PuttySettings.Instance().Sessions)
             {
-                this.sessions.Add(Session.Load(sessionName));
+                Session session;
+                session = Session.Load(sessionName);
+                this.sessions.Add(session);
+                if (session.AutoStart && !session.IsOpen)
+                {
+                    // auto-start session on program startup
+                    try
+                    {
+                        session.Open();
+                    }
+                    catch (SessionAlreadyOpenException)
+                    {
+                        MessageBox.Show("Session already open.");

```

Comment:
MessageBox maybe annoying if something went wrong.
I want to propose use **UserNotifications.Notify(this.session.Name, "<message>");** that i want to design special for any user notifications.